src/app/event-channel/event-channel.service.ts

```diff
@@ -0,0 +1,43 @@
+import { EventEmitter, Injectable } from '@angular/core'
+import { IMqttMessage, IMqttServiceOptions } from 'ngx-mqtt'
+import { BehaviorSubject } from 'rxjs'
+import { EventChannel } from 'src/models/models'
+
+@Injectable({
+  providedIn: 'root'
+})
+
+export class eventChannelService {
+  storedData = localStorage.getItem('oact_telemetry')
+  private readonly cacheData = this.storedData ? JSON.parse(this.storedData) : []
+  public eventChannelLogs: EventChannel = { data: this.cacheData }
+
+  messageSource = new BehaviorSubject<any>(this.eventChannelLogs.data)
+  currentMessage = this.messageSource.asObservable()
+  reconnectMqttService: EventEmitter<IMqttServiceOptions> = new EventEmitter<IMqttServiceOptions>()
+
+  connectionStatusSource = new BehaviorSubject<number>(0)
+  getConnectionStatus = this.connectionStatusSource.asObservable()
+
+  processMessage (message: IMqttMessage): void {
+    this.eventChannelLogs.data.splice(0, 0, JSON.parse(message.payload.toString()))
+    if (this.eventChannelLogs.data.length > 100) {
```

Comment:
not sure this condition is covered base on the codecov.io

Reply:
done